src/main/java/com/esri/samples/search/offline_geocode/OfflineGeocodeSample.java

```diff
@@ -0,0 +1,385 @@
+
+      // add example address suggestions
+      String[] recent = {
+          "1455 Market St, San Francisco, CA 94103", "2011 Mission St, San Francisco  CA  94110",
+          "820 Bryant St, San Francisco  CA  94103", "1 Zoo Rd, San Francisco, 944132",
+          "1201 Mason Street, San Francisco, CA 94108", "151 Third Street, San Francisco, CA 94103",
+          "1050 Lombard Street, San Francisco, CA 94109"
+      };
+      searchBox.getItems().addAll(recent);
+
+      // event to get geocode when query is submitted
+      searchBox.setOnAction((e) -> {
+
+        // get the user's query
+        String query = "";
+        if (searchBox.getSelectionModel().getSelectedIndex() == -1) {
+          // user supplied their own query
+          query = searchBox.getEditor().getText();
+        } else {
+          // user chose a suggested query
+          query = searchBox.getSelectionModel().getSelectedItem();
+        }
+
+        if (!query.equals("")) {
+          // hide callout if showing
+          mapView.getCallout().dismiss();
+
+          // run the locatorTask geocode task
+          ListenableFuture<List<GeocodeResult>> results = locatorTask.geocodeAsync(query, geocodeParameters);
+
+          // add a listener to display the result when loaded
+          results.addDoneListener(new ResultsLoadedListener(results));
+        }
+      });
+
+      // create a basemap from a local tile package
+      TileCache tileCache = new TileCache(System.getProperty("user.dir") +
+          "/samples-data/sanfrancisco/SanFrancisco.tpk");
+      tiledLayer = new ArcGISTiledLayer(tileCache);
+      Basemap basemap = new Basemap(tiledLayer);
+
+      // create ArcGISMap with imagery basemap
+      map = new ArcGISMap(basemap);
+
+      // create a view and set ArcGISMap to it
+      mapView = new MapView();
+      mapView.setMap(map);
+
+      // add a graphics overlay
+      graphicsOverlay = new GraphicsOverlay();
+      graphicsOverlay.setSelectionColor(0xFF00FFFF);
+      mapView.getGraphicsOverlays().add(graphicsOverlay);
+
+      // create a pin graphic
+      Image img = new Image(getClass().getResourceAsStream("/symbols/pin.png"), 0, 80, true, true);
+      pinSymbol = new PictureMarkerSymbol(img);
+      pinSymbol.loadAsync();
+
+      tiledLayer.addDoneLoadingListener(() -> {
+        if (tiledLayer.getLoadStatus() == LoadStatus.LOADED) {
+          Platform.runLater(() -> {
+            // center the view over the tiled layer's extent
+            mapView.setViewpointGeometryAsync(tiledLayer.getFullExtent());
+          });
+        } else {
+          Alert alert = new Alert(Alert.AlertType.ERROR, "Tiled Layer Failed to Load!");
+          alert.show();
+        }
+      });
+
+      // set the callout's default style
+      Callout callout = mapView.getCallout();
+      callout.setLeaderPosition(LeaderPosition.BOTTOM);
+
+      // create a locator task
+      locatorTask = new LocatorTask(System.getProperty("user.dir") +
```

Comment:
Could use code below since that is how we usually do it.
```java
final String locatorPath = new File("samples-data/sanfrancisco/SanFranciscoLocator.loc").getAbsolutePath();
      locatorTask = new LocatorTask(locatorPath);
```